msgpack-core/src/main/java/org/msgpack/value/ValueFactory.java

```diff
@@ -87,12 +87,32 @@ public static ImmutableFloatValue newFloat(double v)
 
     public static ImmutableBinaryValue newBinary(byte[] b)
     {
-        return new ImmutableBinaryValueImpl(b);
+        return newBinary(b, false);
+    }
+
+    public static ImmutableBinaryValue newBinary(byte[] b, boolean ref)
```

Comment:
The meaning of `ref` is a little bit ambiguous. How about using `boolean createCopy`?